packages/at_persistence_secondary_server/lib/src/log/commitlog/commit_log_keystore.dart

```diff
@@ -315,15 +315,41 @@ class CommitLogKeyStore
           if (element.key >= startKey &&
               _acceptKey(element.atKey, regexString) &&
               changes.length <= limit) {
-            changes.add(element);
+            switch (enableCommitId) {
```

Comment:
How about replacing line no. 313 to 352 as below:

```dart
// If limit is null, assign it to max int number.
      limit ??= double.maxFinite as int;
      for (CommitEntry element in values) {
        if (element.key >= startKey &&
            _acceptKey(element.atKey!, regexString) &&
            changes.length <= limit) {
          // On the client side, add the entries whose commit Id's are null
          // Checking the enableCommitId is false to ensure the commit log instance is of client
          if (enableCommitId == false) {
            changes.any((element) => element.commitId == null);
          } else {
            changes.add(element);
          }
        }
      }
      return changes;
```

Reply:
this snipped has been included in the code as part of https://github.com/atsign-foundation/at_server/pull/1020/commits/0d4248ef5abbe50ca1f9f76ae30a08df354fb527